src/Functions/Regexps.h

```diff
@@ -113,12 +113,36 @@ namespace MultiRegexps
         ScratchPtr scratch;
     };
 
+    class RegexpsConstructor
+    {
+    public:
+        RegexpsConstructor() = default;
+
+        void setConstructor(std::function<Regexps()> constructor_) { constructor = std::move(constructor_); }
+
+        Regexps * operator()()
+        {
+            if (regexp)
```

Comment:
@amosbird this code is not thread safe regexp can be read and initialised by other thread during `regexp = constructor();`.